src/components/cms/Dashboard/MyAnalytics.js

```diff
@@ -26,24 +28,35 @@ class MyAnalytics extends Component {
       skillUsageCount: 0,
     };
   }
+
   componentDidMount() {
+    this._isMounted = true;
+
     this.loadSkillsUsage();
   }
 
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
   loadSkillsUsage = async () => {
     const { email, actions } = this.props;
     // eslint-disable-next-line
     try {
       // eslint-disable-next-line camelcase
       let payload = await fetchSkillsByAuthor({ author_email: email });
       this.saveUsageData(payload.authorSkills || []);
-      this.setState({
-        loading: false,
-      });
+      if (this._isMounted) {
+        this.setState({
+          loading: false,
```

Comment:
@akshatnitd the warning is given by this too.. I don't think it's a good practice to bring loading state too into redux store or parent

Reply:
You may visit #3473 for more